src/Money.php

```diff
@@ -61,14 +62,30 @@ public function __construct($amount, Currency $currency)
     }
 
     /**
+     * @param float|string $amount
+     * @param Currency $currency
+     * @return static
+     */
+    public static function fromFloat($amount, Currency $currency)
+    {
+        if (!is_numeric($amount)) {
```

Comment:
`is_numeric` supports "big-numbers" (source: http://php.net/manual/en/function.is-numeric.php#102011)